include/highfive/bits/H5Annotate_traits_misc.hpp

```diff
@@ -25,14 +25,14 @@ inline Attribute
 AnnotateTraits<Derivate>::createAttribute(const std::string& attribute_name,
                                           const DataSpace& space,
                                           const DataType& dtype) {
-    Attribute attribute;
-    if ((attribute._hid = H5Acreate2(
-             static_cast<Derivate*>(this)->getId(), attribute_name.c_str(),
-             dtype._hid, space._hid, H5P_DEFAULT, H5P_DEFAULT)) < 0) {
+    auto attr_id = H5Acreate2(static_cast<Derivate*>(this)->getId(),
+                              attribute_name.c_str(),
+                              dtype._hid, space._hid, H5P_DEFAULT, H5P_DEFAULT);
+    if (attr_id < 0) {
         HDF5ErrMapper::ToException<AttributeException>(
             std::string("Unable to create the attribute \"") + attribute_name + "\":");
     }
-    return attribute;
+    return Attribute(attr_id);
```

Comment:
nicier !

Reply:
I had to do it anyway since constructing an empty Attribute would lead to problems (since it can't get the parent).
That makes me think we should delete empty constructor in these cases.